clients/src/main/java/org/apache/kafka/common/serialization/Serialization.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership. The ASF licenses this file
+ * to You under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.kafka.common.serialization;
+
+/**
+ * The interface for wrapping a serializer and deserializer for the given data type.
+ *
+ * @param <T>
+ */
+public interface Serialization<T> {
```

Comment:
What about naming this interface `Serde` for ser-/deserializer?  @ijuma has a [similar comment])https://github.com/apache/kafka/pull/1066/files#r56135210) down below.

"Serde" is not a totally established term, but I think it captures more accurately the fact that we combine (and thus violate SRP somewhat, cough :-P) serializer and deserializer functionality into one interface.